cmake/Modules/CompileFeature.cmake

```diff
@@ -1,22 +0,0 @@
-# Helper function for coupling add_feature_info(), option(), and add_definitions()
```

Comment:
Is it hard to extract this change into separate PR?
I mean removing this file and using `feature_option` instead of `optional_compile_definitions`?

Reply:
@glassez 

Too complex to factor this out IMO, if you don't want it too be changed at all in this PR. It actually sets global compile definitions via directory-level properties which is very incompatible with the new architecture.